Самообновление списка покупок после удаления и добавления объекта пок…

KraBr0 opened the pull request:
…упок

KraBr0 committed:
Самообновление списка покупок после удаления и добавления объекта покупок

KraBr0 committed:
В акитвити мэйн добавили макет для отображения списков.
Создали два типа итем шоп , активный и не активный

KraBr0 committed:
Установили скролд по списку.
Настроили отоброжение списка. С возмодностью изменения состояния без потери расположения положения в списке.

KraBr0 committed:
Создали ШопЛист Адаптер
для ресайклервью
это нужно для того чтобы, ОС создавало не все элементы , а только те которые видно на экране и + запасные.

dyomin-ea commented on ShoppinhList/app/src/main/java/com/example/shopping_list/data/ShopListRepositoryImpl.kt:
```diff
@@ -1,15 +1,19 @@
 
 object ShopListRepositoryImpl : ShopListRepository {
-    private val shopList = mutableListOf<ShopItem>()
+    private val shopList = sortedSetOf<ShopItem>({ p0, p1 -> p0.id.compareTo(p1.id) }
```
почему именно здесь расположена логика сортировки? почему это не интерактор/юзкейс?

dyomin-ea commented on ShoppinhList/app/src/main/java/com/example/shopping_list/data/ShopListRepositoryImpl.kt:
```diff
@@ -1,15 +1,19 @@
 import com.example.shopping_list.domain.ShopItem
 import com.example.shopping_list.domain.ShopListRepository
 
 object ShopListRepositoryImpl : ShopListRepository {
```
почему репа - синглетон?

dyomin-ea commented on ShoppinhList/app/src/main/java/com/example/shopping_list/data/ShopListRepositoryImpl.kt:
```diff
@@ -1,15 +1,19 @@
+    private val shopList = sortedSetOf<ShopItem>({ p0, p1 -> p0.id.compareTo(p1.id) }
+    )
     private var autoIncrementId=0
+    private val shopListLD= MutableLiveData<List<ShopItem>>()
```
LiveData - это платформенный функционал, завязанный на ЖЦ. Зачем список на нее заменил?

dyomin-ea commented on ShoppinhList/app/src/main/java/com/example/shopping_list/data/ShopListRepositoryImpl.kt:
```diff
@@ -36,8 +42,11 @@ object ShopListRepositoryImpl : ShopListRepository {
 
-    override fun getShopList(): List<ShopItem> {
-        return shopList.toList()
+    override fun getShopList(): LiveData<List<ShopItem>> {
```
почему бы просто свойство наружу не отдать? смысл геттеров в сокрытии какой-то логики получения данных, свойства целиком эту историю закрывают
или есть другие причины скрывать свойство?

dyomin-ea commented on ShoppinhList/app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,6 @@
     <string name="app_name">Shopping_List</string>
+    <string name="add_new_shop_item">Add new shop item</string>
+    <string name="calfetki">Calfetki</string>
+    <string name="_1">1</string>
```
кто это? как я могу понять при чтении, что это за строка?

dyomin-ea commented on ShoppinhList/app/src/main/res/layout/item_shop_enabled.xml:
```diff
@@ -0,0 +1,42 @@
+            android:id="@+id/tv_name"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_weight="1"
```
вес заставляет дважды пересчитать размеры - здесь не принципиально, но это принято плохой практикой
ты же уже знаком с ConstraintLayout, воспользуйся

dyomin-ea commented on ShoppinhList/app/src/main/res/layout/activity_main.xml:
```diff
@@ -6,13 +6,42 @@
+        android:focusable="true"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        android:src="@color/black" />
```
большая круглая черная кнопка? 😅

dyomin-ea commented on ShoppinhList/app/src/main/res/layout/activity_main.xml:
```diff
@@ -6,13 +6,42 @@
 
-    <TextView
-        android:layout_width="wrap_content"
+    <ScrollView
```
а это кто и зачем?

dyomin-ea commented on ShoppinhList/app/src/main/java/com/example/shopping_list/presentaition/ShopListAdepter.kt:
```diff
@@ -0,0 +1,38 @@
+    override fun onBindViewHolder(viewHolder: ShopItemViewHolder, position: Int) {
+        val shopItem=list[position]
+        viewHolder.tvName.text=shopItem.name
+        viewHolder.tvCount.text= shopItem.count.toString()
```
смотри, у тебя ответственность за текст разделена - в ресурсах дефолтное значение для чего-то и кто-то еще сюда постит значения, это не есть хорошо (первое кажется лишним)

dyomin-ea commented on ShoppinhList/app/src/main/java/com/example/shopping_list/presentaition/ShopListAdepter.kt:
```diff
@@ -0,0 +1,38 @@
+
+class ShopListAdepter : RecyclerView.Adapter<ShopListAdepter.ShopItemViewHolder>() {
+
+    val list = listOf<ShopItem>()
```
всегда пустой список? ты точно все реализовал, что хотел?

dyomin-ea commented on ShoppinhList/app/src/main/java/com/example/shopping_list/presentaition/MainViewModel.kt:
```diff
@@ -17,21 +17,16 @@ class MainViewModel: ViewModel() {
     fun deleteShopItem(shopItem: ShopItem){
         deleteShopItemUseCase.deleteShopItem(shopItem)
-        getShopList()
+
```
дырка

KraBr0 committed:
написали РесайклерВью для отображения списка элементов.
При изменении любых параметтров объекта функ onBindViewHolder будет использоваться не для всех элементов списка а только для измененнного за счет сравнения сравнения списков(old,new)через CallBack

KraBr0 committed:
Создали Экран для создания покупки и привезали ее к макету
Создали ВМ гду прописали логику заполнения полей

KraBr0 committed:
установка intent
через фабричный метод newIntent
вход в режим добовления нового эл через кнопку Add
и вход в режим редактировнаия через любой объект

KraBr0 committed:
попытка исправления  отображения статуса

KraBr0 committed:
Егор красава дел было в совпадание фона

KraBr0 committed:
Создан фрагмент.
теперь вью создаются во фрагменте

KraBr0 committed:
переписан метод парс интент, и нью инстаннс для корректого отображения в перевернутом виде

KraBr0 committed:
Работа с бэкстэком
пересоздание фрагмента при перевароте экрана

KraBr0 committed:
land activity
